projects/schematics/src/migrations/2_0/ssr/ssr.ts

```diff
@@ -0,0 +1,298 @@
+    removeFullLineFromImportDeclarationsByModuleName(mainServerTsSourceFile, recorder, '@spartacus/core');
+    recorder.remove(fileContent.indexOf(exportLineToRemove), exportLineToRemove.length);
+
+    tree.commitUpdate(recorder);
+  };
+}
+
+export function removeMapLoaderModule() {
+  return (tree: Tree, context: SchematicContext) => {
+    context.logger.info(
+      'Removing MapLoaderModule import'
+    );
+
+    const mapLoaderModuleTsPath = 'src/app/app.server.module.ts';
+    const buffer = tree.read(mapLoaderModuleTsPath);
+
+    if (!buffer) {
+      throw new SchematicsException('Could not find app.server.module.ts');
+    }
+
+    const recorder = tree.beginUpdate(mapLoaderModuleTsPath);
+    const moduleMapLoaderToBeRemoved = '@nguniversal/module-map-ngfactory-loader';
+    const appServerModuleSourceFile = createSourceFileWithStrippedBOM(tree, mapLoaderModuleTsPath);
+    const printer = ts.createPrinter();
+
+    removeFullLineFromImportDeclarationsByModuleName(appServerModuleSourceFile, recorder, moduleMapLoaderToBeRemoved);
+
+    getDecoratorMetadata(appServerModuleSourceFile, 'NgModule', '@angular/core')
+      .forEach((metadata: ts.ObjectLiteralExpression) => {
+        const matchingProperties = getMetadataField(metadata, 'imports');
+
+        if (!matchingProperties) {
+          return;
+        }
+
+        const assignment = matchingProperties[0] as ts.PropertyAssignment;
+        if (!ts.isArrayLiteralExpression(assignment.initializer)) {
+          return;
+        }
+
+        const arrayLiteral = assignment.initializer;
+        const newImports = arrayLiteral.elements
+          .filter(n => !(ts.isIdentifier(n) && n.text === 'ModuleMapLoaderModule'));
+
+        if (arrayLiteral.elements.length !== newImports.length) {
+          const newImportsText = printer.printNode(
+            ts.EmitHint.Unspecified,
+            ts.updateArrayLiteral(arrayLiteral, newImports),
+            appServerModuleSourceFile,
+          );
+
+          const index = arrayLiteral.getStart();
+          const length = arrayLiteral.getWidth();
+
+          recorder
+            .remove(index, length)
+            .insertLeft(index, newImportsText);
+        }
+      });
+
+    tree.commitUpdate(recorder);
+  };
+}
+
+export function updateAngularJsonFile() {
+  return async (tree: Tree, context: SchematicContext) => {
+    context.logger.info('Updating angular.json builds dist configuration');
+    const projectName = await getProjectNameFromWorkspace(tree);
+
+    const buffer = tree.read('angular.json');
+    if (!buffer) {
+      throw new SchematicsException('Could not find angular.json');
+    }
+
+    const angularJson = JSON.parse(buffer.toString('utf-8'));
+    angularJson.projects[projectName].architect.build.options[
+      'outputPath'
+    ] = `dist/${projectName}/browser`;
+    angularJson.projects[projectName].architect['server'] = {
```

Comment:
Just wondering, is there a need to perform some defensive checks before assuming that SSR is already configured in `angular.json`?

Reply:
No need to, `architect` property is always there, even for clean version of an angular app